bin/cliain/src/commands.rs

```diff
@@ -0,0 +1,238 @@
+use aleph_client::BlockNumber;
+use clap::{Args, Subcommand};
+use primitives::Balance;
+use sp_core::H256;
+use std::path::PathBuf;
+use substrate_api_client::AccountId;
+
+#[derive(Debug, Clone, Args)]
+pub struct ContractOptions {
+    /// balance to transfer from the call origin to the contract
+    #[clap(long, default_value = "0")]
+    pub balance: u128,
+    /// The gas limit enforced when executing the constructor
+    #[clap(long, default_value = "1_000_000_000")]
+    pub gas_limit: u64,
+    /// The maximum amount of balance that can be charged/reserved from the caller to pay for the storage consumed
+    #[clap(long)]
+    pub storage_deposit_limit: Option<u128>,
+}
+
+#[derive(Debug, Clone, Args)]
+pub struct ContractUploadCode {
+    /// Path to the .wasm artifact
+    #[clap(long, parse(from_os_str))]
+    pub wasm_path: PathBuf,
+    /// The maximum amount of balance that can be charged/reserved from the caller to pay for the storage consumed
+    #[clap(long)]
+    pub storage_deposit_limit: Option<u128>,
+}
+
+#[derive(Debug, Clone, Args)]
+pub struct ContractInstantiateWithCode {
+    /// Path to the .wasm artifact
+    #[clap(long, parse(from_os_str))]
+    pub wasm_path: PathBuf,
+    /// Path to the .json file with contract metadata (abi)
+    #[clap(long, parse(from_os_str))]
+    pub metadata_path: PathBuf,
+    /// The name of the contract constructor to call
+    #[clap(name = "constructor", long, default_value = "new")]
```

Comment:
is `constructor` some keyword that you have to specify its name here?

Reply:
likely not, removed